pkg/common-controller/snapshot_controller.go

```diff
@@ -1303,6 +1285,16 @@ func (ctrl *csiSnapshotCommonController) removeSnapshotFinalizer(snapshot *crdv1
 		return nil
 	}
 
+	// If we are here, it means we are going to remove finalizers from the snapshot so that the snapshot can be deleted.
+	// We need to check if there is still PVC finalizer that needs to be removed before removing the snapshot finalizers.
+	// Once snapshot is deleted, there won't be any snapshot update event that can trigger the PVC finalizer removal.
+	if err := ctrl.checkandRemovePVCFinalizer(snapshot, true); err != nil {
```

Comment:
I am not sure this is the best place to put the fix.
checkandRemoveSnapshotFinalizersAndCheckandDeleteContent, which is the caller of this function, has quite some returning points, i.e., this:
https://github.com/kubernetes-csi/external-snapshotter/blob/521502eef991ae09ecd8590cfa67397c7c7da973/pkg/common-controller/snapshot_controller.go#L258

if any of those points is hit, it will still cause the PVC not deletable.